src/Pim/Bundle/CatalogBundle/Elasticsearch/Filter/AbstractAttributeFilter.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+namespace Pim\Bundle\CatalogBundle\Elasticsearch\Filter;
+
+use Akeneo\Component\StorageUtils\Exception\InvalidPropertyException;
+use Pim\Component\Catalog\Model\AttributeInterface;
+use Pim\Component\Catalog\Query\Filter\AttributeFilterInterface;
+use Pim\Component\Catalog\Validator\AttributeValidatorHelper;
+
+/**
+ * Basic implementation of an attribute filter.
+ *
+ * @author    Samir Boulil <samir.boulil@akeneo.com>
+ * @copyright 2017 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+abstract class AbstractAttributeFilter extends AbstractFilter implements AttributeFilterInterface
+{
+    /** @var AttributeValidatorHelper */
+    protected $attrValidatorHelper;
+
+    /** @var string[] */
+    protected $supportedAttributeTypes;
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getAttributeTypes()
+    {
+        return $this->supportedAttributeTypes;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function supportsAttribute(AttributeInterface $attribute)
+    {
+        return in_array($attribute->getType(), $this->supportedAttributeTypes);
+    }
+
+    /**
+     * Check locale and scope are valid
+     *
+     * @param AttributeInterface $attribute
+     * @param string             $locale
+     * @param string             $scope
+     *
+     * @throws InvalidPropertyException
+     */
+    protected function checkLocaleAndScope(AttributeInterface $attribute, $locale, $scope)
+    {
+        try {
+            $this->attrValidatorHelper->validateLocale($attribute, $locale);
+            $this->attrValidatorHelper->validateScope($attribute, $scope);
+        } catch (\LogicException $e) {
+            throw InvalidPropertyException::expectedFromPreviousException(
+                $attribute->getCode(),
+                static::class,
+                $e
+            );
+        }
+    }
+
+    /**
+     * Calculates the ES path to a product value indexed in ES.
```

Comment:
add a TODO that we should find a way to do that better
and to hide / centralize all this dark ES magic